### PR TITLE
Playing with threads/threadgroups to distinguish application run

### PR DESCRIPTION
**Pull request by mbriskar**

### CODE REVIEW

File: utils/src/main/java/org/jboss/windup/util/ExecutionStatistics.java [hunk 1]
```diff
@@ -42,11 +44,49 @@ private ExecutionStatistics()
...
+            stats.remove(thread);
+        }
+    }
+
+    public ExecutionStatistics mergeSiblingThreads() {
+        int numberOfSiblingThreads = Thread.currentThread().getThreadGroup().activeCount();
+        Thread[] threads = new Thread[numberOfSiblingThreads];
+        Thread.currentThread().getThreadGroup().enumerate(threads);
+        for(int i=0;i<threads.length;i++) {
+            merge(stats.get(threads[i]));
```
Comment (mbriskar) [inline utils/src/main/java/org/jboss/windup/util/ExecutionStatistics.java]: this should iterate over all the threads that were created underneath the main windup thread.. However I need to check if dead threads are preserved in this enumeration.

**Comment by windupgithubbot1** [comment]
Refer to this link for build results (access rights to CI server needed): 
http://10.3.8.5//job/windup-pr-builder/1106/<h2>Failed Tests: <span class='status-failure'>52</span></h2><h3><a name='org.jboss.windup:windup-tooling-tests' /><a href='http://10.3.8.5/job/windup-pr-builder/1106/org.jboss.windup$windup-tooling-tests/testReport'>org.jboss.windup:windup-tooling-tests</a>: <span class='status-failure'>1</span></h3><ul><li><a href='http://10.3.8.5/job/windup-pr-builder/1106/org.jboss.windup$windup-tooling-tests/testReport/org.jboss.windup.tooling/ExecutionBuilderTest/testExecutionBuilder/'><strong>org.jboss.windup.tooling.ExecutionBuilderTest.testExecutionBuilder</strong></a></li></ul><h3><a name='org.jboss.windup.ast:windup-java-ast-tests' /><a href='http://10.3.8.5/job/windup-pr-builder/1106/org.jboss.windup.ast$windup-java-ast-tests/testReport'>org.jboss.windup.ast:windup-java-ast-tests</a>: <span class='status-failure'>11</span></h3><ul><li><a href='http://10.3.8.5/job/windup-pr-builder/1106/org.jboss.windup.ast$windup-java-ast-tests/testReport/org.jboss.windup.ast.java.test/JavaASTReferenceResolverTest/testSimpleMain/'><strong>org.jboss.windup.ast.java.test.JavaASTReferenceResolverTest.testSimpleMain</strong></a></li><li><a href='http://10.3.8.5/job/windup-pr-builder/1106/org.jboss.windup.ast$windup-java-ast-tests/testReport/org.jboss.windup.ast.java.test/JavaASTReferenceResolverTest/testNonImportedFQCN/'><strong>org.jboss.windup.ast.java.test.JavaASTReferenceResolverTest.testNonImportedFQCN</strong></a></li><li><a href='http://10.3.8.5/job/windup-pr-builder/1106/org.jboss.windup.ast$windup-java-ast-tests/testReport/org.jboss.windup.ast.java.test/JavaASTReferenceResolverTest/testMyAClass/'><strong>org.jboss.windup.ast.java.test.JavaASTReferenceResolverTest.testMyAClass</strong></a></li><li><a href='http://10.3.8.5/job/windup-pr-builder/1106/org.jboss.windup.ast$windup-java-ast-tests/testReport/org.jboss.windup.ast.java.test/JavaASTReferenceResolverTest/testMyBClass/'><strong>org.jboss.windup.ast.java.test.JavaASTReferenceResolverTest.testMyBClass</strong></a></li><li><a href='http://10.3.8.5/job/windup-pr-builder/1106/org.jboss.windup.ast$windup-java-ast-tests/testReport/org.jboss.windup.ast.java.test/JavaASTReferenceResolverTest/testJavaLangReferences/'><strong>org.jboss.windup.ast.java.test.JavaASTReferenceResolverTest.testJavaLangReferences</strong></a></li><li><a href='http://10.3.8.5/job/windup-pr-builder/1106/org.jboss.windup.ast$windup-java-ast-tests/testReport/org.jboss.windup.ast.java.test/JavaASTReferenceResolverTest/testWildcardImport/'><strong>org.jboss.windup.ast.java.test.JavaASTReferenceResolverTest.testWildcardImport</strong></a></li><li><a href='http://10.3.8.5/job/windup-pr-builder/1106/org.jboss.windup.ast$windup-java-ast-tests/testReport/org.jboss.windup.ast.java.test/JavaASTReferenceResolverTest/testInnerClassScenarios/'><strong>org.jboss.windup.ast.java.test.JavaASTReferenceResolverTest.testInnerClassScenarios</strong></a></li><li><a href='http://10.3.8.5/job/windup-pr-builder/1106/org.jboss.windup.ast$windup-java-ast-tests/testReport/org.jboss.windup.ast.java.test/JavaASTReferenceResolverTest/testInterfaceExtension/'><strong>org.jboss.windup.ast.java.test.JavaASTReferenceResolverTest.testInterfaceExtension</strong></a></li><li><a href='http://10.3.8.5/job/windup-pr-builder/1106/org.jboss.windup.ast$windup-java-ast-tests/testReport/org.jboss.windup.ast.java.test/JavaASTReferenceResolverTest/testHelloWorld/'><strong>org.jboss.windup.ast.java.test.JavaASTReferenceResolverTest.testHelloWorld</strong></a></li><li><a href='http://10.3.8.5/job/windup-pr-builder/1106/org.jboss.windup.ast$windup-java-ast-tests/testReport/org.jboss.windup.ast.java.test/JavaAnnotationScanningTest/testSimpleAnnotatedClass/'><strong>org.jboss.windup.ast.java.test.JavaAnnotationScanningTest.testSimpleAnnotatedClass</strong></a></li><li><a href='http://10.3.8.5/job/windup-pr-builder/1106/org.jboss.windup.ast$windup-java-ast-tests/testReport/org.jboss.windup.ast.java.test/JavaAnnotationScanningTest/testComplexAnnotatedClass/'><strong>org.jboss.windup.ast.java.test.JavaAnnotationScanningTest.testComplexAnnotatedClass</strong></a></li></ul><h3><a name='org.jboss.windup.rules.apps:windup-rules-base-tests' /><a href='http://10.3.8.5/job/windup-pr-builder/1106/org.jboss.windup.rules.apps$windup-rules-base-tests/testReport'>org.jboss.windup.rules.apps:windup-rules-base-tests</a>: <span class='status-failure'>1</span></h3><ul><li><a href='http://10.3.8.5/job/windup-pr-builder/1106/org.jboss.windup.rules.apps$windup-rules-base-tests/testReport/org.jboss.windup.rules.apps.condition/FileContentTest/testFileContentScan/'><strong>org.jboss.windup.rules.apps.condition.FileContentTest.testFileContentScan</strong></a></li></ul><h3><a name='org.jboss.windup.rules.apps:windup-rules-java-tests' /><a href='http://10.3.8.5/job/windup-pr-builder/1106/org.jboss.windup.rules.apps$windup-rules-java-tests/testReport'>org.jboss.windup.rules.apps:windup-rules-java-tests</a>: <span class='status-failure'>15</span></h3><ul><li><a href='http://10.3.8.5/job/windup-pr-builder/1106/org.jboss.windup.rules.apps$windup-rules-java-tests/testReport/org.jboss.windup.rule.groovy/GroovyExtensionJavaRulesTest/testHintsAndClassificationOperation/'><strong>org.jboss.windup.rule.groovy.GroovyExtensionJavaRulesTest.testHintsAndClassificationOperation</strong></a></li><li><a href='http://10.3.8.5/job/windup-pr-builder/1106/org.jboss.windup.rules.apps$windup-rules-java-tests/testReport/org.jboss.windup.rules.apps.java/ClassMetadataTest/testJavaSourceScanning/'><strong>org.jboss.windup.rules.apps.java.ClassMetadataTest.testJavaSourceScanning</strong></a></li><li><a href='http://10.3.8.5/job/windup-pr-builder/1106/org.jboss.windup.rules.apps$windup-rules-java-tests/testReport/org.jboss.windup.rules.apps.java/ClassMetadataTest/testJavaClassFiles/'><strong>org.jboss.windup.rules.apps.java.ClassMetadataTest.testJavaClassFiles</strong></a></li><li><a href='http://10.3.8.5/job/windup-pr-builder/1106/org.jboss.windup.rules.apps$windup-rules-java-tests/testReport/org.jboss.windup.rules.apps.java.scan.ast/ASTProcessorIntegrationTest/testJavaSourceScanning/'><strong>org.jboss.windup.rules.apps.java.scan.ast.ASTProcessorIntegrationTest.testJavaSourceScanning</strong></a></li><li><a href='http://10.3.8.5/job/windup-pr-builder/1106/org.jboss.windup.rules.apps$windup-rules-java-tests/testReport/org.jboss.windup.rules.java/GatherIgnoredFilesTest/testJavaClassCondition/'><strong>org.jboss.windup.rules.java.GatherIgnoredFilesTest.testJavaClassCondition</strong></a></li><li><a href='http://10.3.8.5/job/windup-pr-builder/1106/org.jboss.windup.rules.apps$windup-rules-java-tests/testReport/org.jboss.windup.rules.java/JavaClassBindingStatusTest/testResolutionStatus/'><strong>org.jboss.windup.rules.java.JavaClassBindingStatusTest.testResolutionStatus</strong></a></li><li><a href='http://10.3.8.5/job/windup-pr-builder/1106/org.jboss.windup.rules.apps$windup-rules-java-tests/testReport/org.jboss.windup.rules.java/JavaClassCompositeTest/testJavaClassCondition/'><strong>org.jboss.windup.rules.java.JavaClassCompositeTest.testJavaClassCondition</strong></a></li><li><a href='http://10.3.8.5/job/windup-pr-builder/1106/org.jboss.windup.rules.apps$windup-rules-java-tests/testReport/org.jboss.windup.rules.java/JavaClassSourceMatchTest/testJavaClassCondition/'><strong>org.jboss.windup.rules.java.JavaClassSourceMatchTest.testJavaClassCondition</strong></a></li><li><a href='http://10.3.8.5/job/windup-pr-builder/1106/org.jboss.windup.rules.apps$windup-rules-java-tests/testReport/org.jboss.windup.rules.java/JavaClassTest/testJavaClassCondition/'><strong>org.jboss.windup.rules.java.JavaClassTest.testJavaClassCondition</strong></a></li><li><a href='http://10.3.8.5/job/windup-pr-builder/1106/org.jboss.windup.rules.apps$windup-rules-java-tests/testReport/org.jboss.windup.rules.java/JavaClassWithoutClassificationTest/testJavaClassCondition/'><strong>org.jboss.windup.rules.java.JavaClassWithoutClassificationTest.testJavaClassCondition</strong></a></li><li><a href='http://10.3.8.5/job/windup-pr-builder/1106/org.jboss.windup.rules.apps$windup-rules-java-tests/testReport/org.jboss.windup.rules.java/JavaClassWithoutHintTest/testJavaClassCondition/'><strong>org.jboss.windup.rules.java.JavaClassWithoutHintTest.testJavaClassCondition</strong></a></li><li><a href='http://10.3.8.5/job/windup-pr-builder/1106/org.jboss.windup.rules.apps$windup-rules-java-tests/testReport/org.jboss.windup.rules.java/JavaClassXmlRulesTest/testJavaClassCondition/'><strong>org.jboss.windup.rules.java.JavaClassXmlRulesTest.testJavaClassCondition</strong></a></li><li><a href='http://10.3.8.5/job/windup-pr-builder/1106/org.jboss.windup.rules.apps$windup-rules-java-tests/testReport/org.jboss.windup.rules.java/JavaHintsClassificationsTest/testHintsAndClassificationOperation/'><strong>org.jboss.windup.rules.java.JavaHintsClassificationsTest.testHintsAndClassificationOperation</strong></a></li><li><a href='http://10.3.8.5/job/windup-pr-builder/1106/org.jboss.windup.rules.apps$windup-rules-java-tests/testReport/org.jboss.windup.rules.java/SourceModeTest/testSourceModePositive/'><strong>org.jboss.windup.rules.java.SourceModeTest.testSourceModePositive</strong></a></li><li><a href='http://10.3.8.5/job/windup-pr-builder/1106/org.jboss.windup.rules.apps$windup-rules-java-tests/testReport/org.jboss.windup.rules.java.ignore/JavaIgnoreRegexesTest/testHintsAndClassificationOperation/'><strong>org.jboss.windup.rules.java.ignore.JavaIgnoreRegexesTest.testHintsAndClassificationOperation</strong></a></li></ul><h3><a name='org.jboss.windup.rules.apps:windup-rules-xml-tests' /><a href='http://10.3.8.5/job/windup-pr-builder/1106/org.jboss.windup.rules.apps$windup-rules-xml-tests/testReport'>org.jboss.windup.rules.apps:windup-rules-xml-tests</a>: <span class='status-failure'>1</span></h3><ul><li><a href='http://10.3.8.5/job/windup-pr-builder/1106/org.jboss.windup.rules.apps$windup-rules-xml-tests/testReport/org.jboss.windup.rules.xml/XmlAndJavaParameterizedTest/testXmlAndJavaSearchParams/'><strong>org.jboss.windup.rules.xml.XmlAndJavaParameterizedTest.testXmlAndJavaSearchParams</strong></a></li></ul><h3><a name='org.jboss.windup.tests:windup-tests' /><a href='http://10.3.8.5/job/windup-pr-builder/1106/org.jboss.windup.tests$windup-tests/testReport'>org.jboss.windup.tests:windup-tests</a>: <span class='status-failure'>18</span></h3><ul><li><a href='http://10.3.8.5/job/windup-pr-builder/1106/org.jboss.windup.tests$windup-tests/testReport/org.jboss.windup.tests.application/IgnoreJavaClassTest/testIgnoreFiles/'><strong>org.jboss.windup.tests.application.IgnoreJavaClassTest.testIgnoreFiles</strong></a></li><li><a href='http://10.3.8.5/job/windup-pr-builder/1106/org.jboss.windup.tests$windup-tests/testReport/org.jboss.windup.tests.application/MavenInfoMissingBinaryModeTest/testRunMavenInfoMissing/'><strong>org.jboss.windup.tests.application.MavenInfoMissingBinaryModeTest.testRunMavenInfoMissing</strong></a></li><li><a href='http://10.3.8.5/job/windup-pr-builder/1106/org.jboss.windup.tests$windup-tests/testReport/org.jboss.windup.tests.application/WindupArchitectureCatchallTest/testRunWindupJsp/'><strong>org.jboss.windup.tests.application.WindupArchitectureCatchallTest.testRunWindupJsp</strong></a></li><li><a href='http://10.3.8.5/job/windup-pr-builder/1106/org.jboss.windup.tests$windup-tests/testReport/org.jboss.windup.tests.application/WindupArchitectureHibernateTest/testRunWindupHibernate/'><strong>org.jboss.windup.tests.application.WindupArchitectureHibernateTest.testRunWindupHibernate</strong></a></li><li><a href='http://10.3.8.5/job/windup-pr-builder/1106/org.jboss.windup.tests$windup-tests/testReport/org.jboss.windup.tests.application/WindupArchitectureJEEExampleTest/testRunWindupJEEExampleMode/'><strong>org.jboss.windup.tests.application.WindupArchitectureJEEExampleTest.testRunWindupJEEExampleMode</strong></a></li><li><a href='http://10.3.8.5/job/windup-pr-builder/1106/org.jboss.windup.tests$windup-tests/testReport/org.jboss.windup.tests.application/WindupArchitectureJspTest/testRunWindupJsp/'><strong>org.jboss.windup.tests.application.WindupArchitectureJspTest.testRunWindupJsp</strong></a></li><li><a href='http://10.3.8.5/job/windup-pr-builder/1106/org.jboss.windup.tests$windup-tests/testReport/org.jboss.windup.tests.application/WindupArchitectureMediumBinaryModeTest/testRunWindupMedium/'><strong>org.jboss.windup.tests.application.WindupArchitectureMediumBinaryModeTest.testRunWindupMedium</strong></a></li><li><a href='http://10.3.8.5/job/windup-pr-builder/1106/org.jboss.windup.tests$windup-tests/testReport/org.jboss.windup.tests.application/WindupArchitectureScanPackagesTest/testRunWindupScanPackages/'><strong>org.jboss.windup.tests.application.WindupArchitectureScanPackagesTest.testRunWindupScanPackages</strong></a></li><li><a href='http://10.3.8.5/job/windup-pr-builder/1106/org.jboss.windup.tests$windup-tests/testReport/org.jboss.windup.tests.application/WindupArchitectureSeamBookingSourceTest/testRunWindupSeamBookingSourceMode/'><strong>org.jboss.windup.tests.application.WindupArchitectureSeamBookingSourceTest.testRunWindupSeamBookingSourceMode</strong></a></li><li><a href='http://10.3.8.5/job/windup-pr-builder/1106/org.jboss.windup.tests$windup-tests/testReport/org.jboss.windup.tests.application/WindupArchitectureSmallBinaryMode2Test/testRunWindupTiny/'><strong>org.jboss.windup.tests.application.WindupArchitectureSmallBinaryMode2Test.testRunWindupTiny</strong></a></li><li><a href='http://10.3.8.5/job/windup-pr-builder/1106/org.jboss.windup.tests$windup-tests/testReport/org.jboss.windup.tests.application/WindupArchitectureSmallBinaryModeTest/testRunWindupTiny/'><strong>org.jboss.windup.tests.application.WindupArchitectureSmallBinaryModeTest.testRunWindupTiny</strong></a></li><li><a href='http://10.3.8.5/job/windup-pr-builder/1106/org.jboss.windup.tests$windup-tests/testReport/org.jboss.windup.tests.application/WindupArchitectureSoa5GatewaysRoutersTest/testRunWindupSourceMode/'><strong>org.jboss.windup.tests.application.WindupArchitectureSoa5GatewaysRoutersTest.testRunWindupSourceMode</strong></a></li><li><a href='http://10.3.8.5/job/windup-pr-builder/1106/org.jboss.windup.tests$windup-tests/testReport/org.jboss.windup.tests.application/WindupArchitectureSourceModeTest/testRunWindupSourceMode/'><strong>org.jboss.windup.tests.application.WindupArchitectureSourceModeTest.testRunWindupSourceMode</strong></a></li><li><a href='http://10.3.8.5/job/windup-pr-builder/1106/org.jboss.windup.tests$windup-tests/testReport/org.jboss.windup.tests.application/WindupArchitectureSourceWithBadPOMTest/testRunWindupSourceMode/'><strong>org.jboss.windup.tests.application.WindupArchitectureSourceWithBadPOMTest.testRunWindupSourceMode</strong></a></li><li><a href='http://10.3.8.5/job/windup-pr-builder/1106/org.jboss.windup.tests$windup-tests/testReport/org.jboss.windup.tests.application/WindupArchitectureSpringSmallTest/testRunWindupSmallSpringApp/'><strong>org.jboss.windup.tests.application.WindupArchitectureSpringSmallTest.testRunWindupSmallSpringApp</strong></a></li><li><a href='http://10.3.8.5/job/windup-pr-builder/1106/org.jboss.windup.tests$windup-tests/testReport/org.jboss.windup.tests.application/WindupCompiledWithSourceTest/testRunWindupOnAppWithJars/'><strong>org.jboss.windup.tests.application.WindupCompiledWithSourceTest.testRunWindupOnAppWithJars</strong></a></li><li><a href='http://10.3.8.5/job/windup-pr-builder/1106/org.jboss.windup.tests$windup-tests/testReport/org.jboss.windup.tests.application/WindupCompiledWithSourceTest/testRunWindupOnAppWithoutJars/'><strong>org.jboss.windup.tests.application.WindupCompiledWithSourceTest.testRunWindupOnAppWithoutJars</strong></a></li><li><a href='http://10.3.8.5/job/windup-pr-builder/1106/org.jboss.windup.tests$windup-tests/testReport/org.jboss.windup.tests.application/WindupCompiledWithSourceTest/testRunWindupOnJarWithSourceAndClassFiles/'><strong>org.jboss.windup.tests.application.WindupCompiledWithSourceTest.testRunWindupOnJarWithSourceAndClassFiles</strong></a></li></ul><h3><a name='org.jboss.windup.ui:windup-ui-tests' /><a href='http://10.3.8.5/job/windup-pr-builder/1106/org.jboss.windup.ui$windup-ui-tests/testReport'>org.jboss.windup.ui:windup-ui-tests</a>: <span class='status-failure'>5</span></h3><ul><li><a href='http://10.3.8.5/job/windup-pr-builder/1106/org.jboss.windup.ui$windup-ui-tests/testReport/org.jboss.windup.addon.ui/WindupCommandTest/testUserRulesDirMigration/'><strong>org.jboss.windup.addon.ui.WindupCommandTest.testUserRulesDirMigration</strong></a></li><li><a href='http://10.3.8.5/job/windup-pr-builder/1106/org.jboss.windup.ui$windup-ui-tests/testReport/org.jboss.windup.addon.ui/WindupCommandTest/testDuplicateUserRulesDirMigration/'><strong>org.jboss.windup.addon.ui.WindupCommandTest.testDuplicateUserRulesDirMigration</strong></a></li><li><a href='http://10.3.8.5/job/windup-pr-builder/1106/org.jboss.windup.ui$windup-ui-tests/testReport/org.jboss.windup.addon.ui/WindupCommandTest/testNewMigration/'><strong>org.jboss.windup.addon.ui.WindupCommandTest.testNewMigration</strong></a></li><li><a href='http://10.3.8.5/job/windup-pr-builder/1106/org.jboss.windup.ui$windup-ui-tests/testReport/org.jboss.windup.addon.ui/WindupCommandTest/testUserIgnoreDirMigration/'><strong>org.jboss.windup.addon.ui.WindupCommandTest.testUserIgnoreDirMigration</strong></a></li><li><a href='http://10.3.8.5/job/windup-pr-builder/1106/org.jboss.windup.ui$windup-ui-tests/testReport/org.jboss.windup.addon.ui/WindupCommandTest/testOutputDefaultValue/'><strong>org.jboss.windup.addon.ui.WindupCommandTest.testOutputDefaultValue</strong></a></li></ul>
Test FAILed.

**Comment by windupgithubbot1** [comment]
Refer to this link for build results (access rights to CI server needed): 
http://10.3.8.5//job/windup-core-PR-dependents-check/475/
Test FAILed.

**Comment by windupgithubbot1** [comment]
Refer to this link for build results (access rights to CI server needed): 
http://10.3.8.5//job/windup-pr-builder/1109/
Test FAILed.

**Comment by windupgithubbot1** [comment]
Refer to this link for build results (access rights to CI server needed): 
http://10.3.8.5//job/windup-core-PR-dependents-check/478/
Test FAILed.

**Comment by windupgithubbot1** [comment]
Refer to this link for build results (access rights to CI server needed): 
http://10.3.8.5//job/windup-core-PR-dependents-check/483/
Test PASSed.

**Comment by windupgithubbot1** [comment]
Refer to this link for build results (access rights to CI server needed): 
http://10.3.8.5//job/windup-pr-builder/1114/
Test PASSed.

**Comment by windupgithubbot1** [comment]
Refer to this link for build results (access rights to CI server needed): 
http://10.3.8.5//job/windup-core-PR-dependents-check/484/
Test PASSed.

**Comment by windupgithubbot1** [comment]
Refer to this link for build results (access rights to CI server needed): 
http://10.3.8.5//job/windup-pr-builder/1115/
Test PASSed.

**Comment by windupgithubbot1** [comment]
Refer to this link for build results (access rights to CI server needed): 
http://10.3.8.5//job/windup-core-PR-dependents-check/485/
Test PASSed.

File: utils/src/main/java/org/jboss/windup/util/ExecutionStatistics.java [hunk 2]
```diff
@@ -42,11 +46,52 @@ private ExecutionStatistics()
...
+    }
+
+    public void clear(Iterable<Thread> threads) {
+        for (Thread thread : threads)
+        {
+            stats.remove(thread);
+        }
+    }
+    
+    public void merge() {
```
Comment (mbriskar) [inline utils/src/main/java/org/jboss/windup/util/ExecutionStatistics.java]: this must be synchronized

**Comment by jsight** [comment]
retest

**Comment by windupgithubbot1** [comment]
Refer to this link for build results (access rights to CI server needed): 
http://wonka.mw.lab.eng.bos.redhat.com:9080/jenkins//job/windup-core-PR-dependents-check/497/
Test PASSed.

**Comment by windupgithubbot1** [comment]
Refer to this link for build results (access rights to CI server needed): 
http://wonka.mw.lab.eng.bos.redhat.com:9080/jenkins//job/windup-pr-builder/1127/
Test PASSed.

File: utils/src/test/java/org/jboss/windup/util/ExecutionStatisticsTest.java [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+
+import java.util.Map;
+import java.util.concurrent.Callable;
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.TimeUnit;
+
+/**
+ * Testing ExecutionStatistics in a multi-threaded environment
+ */
+public class ExecutionStatisticsTest
```
Comment (jsight) [inline utils/src/test/java/org/jboss/windup/util/ExecutionStatisticsTest.java]: This test is really nicely done!

File: utils/src/main/java/org/jboss/windup/util/ExecutionStatistics.java [hunk 4]
```diff
@@ -42,11 +46,52 @@ private ExecutionStatistics()
...
+            stats.put(currentThread,new ExecutionStatistics());
+        }
+        return stats.get(currentThread);
+    }
+
+    public Map<String, TimingData> getExecutionInfo() {
+        return executionInfo;
+    }
+
+    public void clear(Iterable<Thread> threads) {
```
Comment (jsight) [inline utils/src/main/java/org/jboss/windup/util/ExecutionStatistics.java]: This doesn't appear to be called from anywhere. That seems like a mistake?

File: utils/src/main/java/org/jboss/windup/util/threading/WindupExecutors.java [hunk 5]
```diff
@@ -0,0 +1,20 @@
...
+import java.util.concurrent.Executors;
+
+/**
+ * An alterantive to {@link Executors} that pushes windup specific threadFactory inside by default. This is useful in order to have
+ * be able to connect sibling threads into the main thread of windup run.
+ */
+public class WindupExecutors
+{
+    public static ExecutorService newFixedThreadPool(int numberOfThreads) {
+        return Executors.newFixedThreadPool(numberOfThreads,new WindupChildThreadFactory());
```
Comment (jsight) [inline utils/src/main/java/org/jboss/windup/util/threading/WindupExecutors.java]: IMO, a custom ExecutorService would be better than a custom threadfactory. The service will reuse created threads, which will be more leak-prone.

This is more than good enough for now, though. :)

**Comment by jsight** [comment]
Other than the question regarding "clear", this is good to merge.

**Comment by windupgithubbot1** [comment]
Refer to this link for build results (access rights to CI server needed): 
http://wonka.mw.lab.eng.bos.redhat.com:9080/jenkins//job/windup-core-PR-dependents-check/511/

File: utils/src/main/java/org/jboss/windup/util/threading/WindupExecutors.java (hunk 5 above)
Comment (mbriskar) [inline utils/src/main/java/org/jboss/windup/util/threading/WindupExecutors.java]: I am open to any suggestions. However isn't it better to discard all the threads once we are done with the work? And then recreate it?

File: utils/src/test/java/org/jboss/windup/util/ExecutionStatisticsTest.java (hunk 3 above)
Comment (mbriskar) [inline utils/src/test/java/org/jboss/windup/util/ExecutionStatisticsTest.java]: thanks :)

**Comment by windupgithubbot1** [comment]
Refer to this link for build results (access rights to CI server needed): 
http://wonka.mw.lab.eng.bos.redhat.com:9080/jenkins//job/windup-pr-builder/1140/

**Comment by windupgithubbot1** [comment]
Refer to this link for build results (access rights to CI server needed): 
http://wonka.mw.lab.eng.bos.redhat.com:9080/jenkins//job/windup-core-PR-dependents-check/512/

**Pull request merged**